kong/global.lua

```diff
@@ -186,16 +186,21 @@ function _GLOBAL.init_worker_events()
     worker_events = require "resty.worker.events"
 
   else
-    local sock_name = "worker_events.sock"
-    if ngx.config.subsystem == "stream" then
-      sock_name = "stream_" .. sock_name
-    end
+    -- `kong.configuration.prefix` is already normalized to an absoluate path,
+    -- but `ngx.config.prefix()` is not
+    local prefix = configuration and
+                   configuration.resty_events_sock_path or configuration.prefix or
+                   require("pl.path").abspath(ngx.config.prefix())
+
+    local sock = ngx.config.subsystem == "stream" and
+                 "stream_worker_events.sock" or "worker_events.sock"
+
+    local listening = "unix:" .. prefix .. "/" .. sock
```

Comment:
can't we construct this path in the config loader module? and store it in `resty_events_sock_path`. From there on, everywhere else we need it we can just grab it from the config, without have to do the construction again.

Would also mean it is reported on the management api, as the full filename, without a user having to apply logic where it was created.

Reply:
sounds good.

Reply:
resty-events library needs two socket file, one in http subsystem, another in stream subsystem, so we can only set a path, not a file name.